src/aligners/nw.rs

```diff
@@ -20,6 +21,119 @@ const INF: Cost = Cost::MAX / 2;
 type NWLayers<const N: usize> = Layers<N, Vec<Cost>>;
 
 impl<const N: usize> NW<AffineCost<N>> {
+    fn track_path(&self, layers: &mut Vec<NWLayers<N>>, a: &Sequence, b: &Sequence) -> PATH {
+        let mut path: PATH = vec![(layers.len(), layers[0].m.len())];
+        let mut x = layers.len();
+        let mut y = layers[0].m.len();
+        let mut t: isize = -1;
+        let mut save = |x: &usize, y: &usize| path.push((*x, *y));
+
+        while x + y > 0 {
+            let mut found = false;
+            if t == -1 {
+                if x * y > 0 {
+                    //x*y >0 == (x > 0 && y > 0)
+                    if a[x - 1] == b[y - 1] && layers[x].m[y] == layers[x - 1].m[y - 1] {
+                        //match
+                        x -= 1;
+                        y -= 1;
+                        save(&x, &y);
+                        found = true;
+                    }
+                    if !found {
+                        if let Some(sub) = self.cm.sub {
+                            if layers[x].m[y] == layers[x - 1].m[y - 1] + sub {
+                                //mismatch
+                                x -= 1;
+                                y -= 1;
+                                save(&x, &y);
+                                found = true;
+                            }
+                        }
+                    }
+                }
+                if x > 0 && !found {
+                    if let Some(ins) = self.cm.ins {
+                        if layers[x].m[y] == layers[x - 1].m[y] + ins {
+                            //insertion
+                            x -= 1;
+                            save(&x, &y);
+                            found = true;
+                        }
+                    }
+                }
+                if y > 0 && !found {
+                    if let Some(del) = self.cm.del {
+                        if layers[x].m[y] == layers[x].m[y - 1] + del {
+                            //deletion
+                            y -= 1;
+                            save(&x, &y);
+                            found = true;
+                        }
+                    }
+                }
+                //affine layers check
+                if !found {
+                    let mut i = 0;
+                    for affine_layer in &layers[x].affine {
+                        if layers[x].m[y] == affine_layer[y] {
+                            t = i;
+                            break;
+                        }
+                        i += 1;
+                    }
+                }
+            } else {
+                //Attention! Loop below covers only cases when x > 0
+                if x > 0 {
+                    for (cm, prev_layer, next_layer) in
+                        izip!(&self.cm.affine, &layers[x - 1].affine, &layers[x].affine)
```

Comment:
hmm; this `x=0` case is a bit annoying. Maybe you can make separate for loops for the insert and delete case?